www/index.html

```diff
@@ -189,37 +189,57 @@
+                                </div>
+                                <div id="scanningForArchives" style="display: none;">
+                                    <br /> Scanning for archives... Please wait <img src="img/spinner.gif" alt="Please wait..." />
+                                </div>
+                                <div id="chooseArchiveFromLocalStorage" style="display: none;">
+                                    <br /> Please select the archive you want to use : <select id="archiveList" class="form-control"></select>
+                                    <br /> Click <a id="btnRescanDeviceStorage">here</a> to rescan your SD Cards and internal memory
+                                </div>
+                            </div>
+                        </div>
                     </div>
-                    <div id="scanningForArchives" style="display: none;">
-                        <br /> Scanning for archives... Please wait <img src="img/spinner.gif" alt="Please wait..." />
-                    </div>
-                    <div id="chooseArchiveFromLocalStorage" style="display: none;">
-                        <br /> Please select the archive you want to use : <select id="archiveList" class="form-control"></select>
-                        <br /> Click <a id="btnRescanDeviceStorage">here</a> to rescan your SD Cards and internal memory
+
+                    <div class="container">
+                        <h3>Expert settings</h3>
+                        <div class="row">
+                            <div class="column">
+                                <div class="panel panel-danger" id="contentInjectionModeDiv">
+                                    <div class="panel-heading">Do not touch unless you know what you're doing!</div>
+                                    <div class="panel-body">
+                                        <p>Content injection mode:</p>
+                                        <p>
+                                            <input type="radio" name="contentInjectionMode" value="jquery" id="jQueryModeRadio" checked>
+                                            <b>&nbsp;JQuery</b> (slow and memory hungry, but safer)
+                                        </p>
+                                        <p>
+                                            <input type="radio" name="contentInjectionMode" value="serviceworker" id="serviceworkerModeRadio">
+                                            <b>&nbsp;ServiceWorker</b> (faster but unstable, and not supported by all platforms)
+                                        </p>
+                                    </div>
+                                </div>
+                                <div id="serviceWorkerStatus"></div>
+                                <div id="messageChannelStatus"></div>
```

Comment:
Maybe those two divs could be put inside a panel too, with a heading like "API status"?

Reply:
Good idea, I'll take a look.